spec/models/auth/voms_spec.rb

```diff
@@ -28,6 +28,46 @@
     end
   end
 
+  describe '#parse_hash_groups!' do
+    context 'with invalid groups' do
+      let(:env_hash) do
+        { 'GRST_VOMS_FQANS' => '/coolclub/Role=NULL/Capability=NUL;' \
+                               '/nerds/Roe=LL/Capabili=NLL;' \
```

Comment:
(repost)

This is probably something we haven't talked about at all because I forgot to mention it ... sorry. In VOMS, groups can have subgroups. This would look like `/coolclub/muchcoolersubclub/Role=NULL/Capability=NULL`.

Now, we DO NOT want to support subgroups, let me be clear on that. But we should have a clearly defined behavior when such credentials arrive.

I would suggest just dropping subgroups. That would mean dropping every matched group with `/` in its name (as matched by `%r{^\/(?<group>[^\s]+)\/Role=(?<role>[^\s]+)\/Capability=NULL$}`). Opinions?

Reply:
That sounds reasonable and easy to implement. I'm in.

Reply:
Can you also change the regexp to `%r{^\/(?<group>[^\s]+)\/Role=(?<role>[^\s]+)\/Capability=(?<capability>[^\s]+)$}`? We shouldn't ignore lines with a capability just because we are ignoring capabilities.

Reply:
So basically take all capabilities?

Reply:
Yes. Or just match a line with any capability. There is no need to capture it, I guess.

Reply:
Well, I guess we should avoid `NULL` capability, the same as with the roles. But that has nothing to do with the regexp. Sooo, this comment is totally useless. Carry on...